src/genie/genie_sp_elt.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+"""
+ELT pipeline for sponsored project clinical data.
+Reads YAML config for cohorts, fetches Synapse files, and writes to Snowflake.
+"""
+
+import argparse
+import os
+
+import yaml
+import pandas as pd
+import synapseclient
+
+from snowflake_utils import get_connection, write_to_snowflake, logger
+
+
+def process_cohort(
+    syn: synapseclient.Synapse,
+    conn : "snowflake.connector.SnowflakeConnection",
+    cohort_config: dict,
+    database: str,
+    overwrite: bool,
+):
+    """ Process a single cohort:
+      - Download files from Synapse
+      - Use specified patient_id_key and sample_id_key
+      - Write to Snowflake table 
+
+    Args:
+        syn (synapseclient.Synapse): synapse client connection
+        conn (snowflake.connector.SnowflakeConnection): snowflake connection
+        cohort_config (dict): cohort config containing synapse id of the files to ingest
+        database (str): name of the database to ingest to
+        overwrite (bool): whether to overwrite the table or not
+    """
+    schema_name = cohort_config["cohort"]
+    table_name = cohort_config["table_name"]
+    file_synids = cohort_config["file_synid"]
+
+    logger.info(f"Creating/using schema: {schema_name}")
+
+    with conn.cursor() as cs:
+        cs.execute(f"USE DATABASE {database};")
+        cs.execute(
+            f"CREATE SCHEMA IF NOT EXISTS {schema_name} WITH MANAGED ACCESS;"
```

Comment:
Consider saving each sponsored project's clinical file as a table under a single schema called SPONSORED_PROJECTS or SP since each sponsored project doesn't have versions or multiple files that we want to ingest.

Reply:
^ Can be a separate ticket if this ends up being a lot of work